Issue with JSX runtime

I noticed an error when playing with the provided example. The only changes I made were in the package.json setting the dependency for `@stylex-extend/react` to whatever the last is on NPM because it was `workspace:*` or something before that. 

Here is the error:

`Cannot find module '@stylex-extend/react/jsx-runtime' or its corresponding type declarations.ts(2307)`

Repro here: https://github.com/zaydek/nonzzz-stylex-extend

<img width="830" alt="Screenshot 2024-05-21 at 2 53 58 AM" src="https://github.com/nonzzz/stylex-extend/assets/58870766/ffc194e0-4301-4c63-bfe0-3801930aea28">

--- a/packages/react/package.json
+++ b/packages/react/package.json
@@ -16,12 +16,22 @@
     ".": "./src/index.js",
     "./jsx-runtime": {
       "import": "./src/jsx-runtime.mjs",
-      "require": "./src/jsx-runtime.js"
+      "require": "./src/jsx-runtime.js",
+      "types":"./src/jsx-dev-runtime.d.ts"
     },
     "./jsx-dev-runtime.js": {
       "import": "./src/jsx-dev-runtime.mjs",
-      "require": "./src/jsx-dev-runtime.js"
+      "require": "./src/jsx-dev-runtime.js",
+      "types":"./src/jsx-dev-runtime.d.ts"
     },
-    "./*": "./src/*"
+    "./*": "./src/*",
+    "./package.json":"./package.json"
+  },
+  "typesVersions": {
+    "*":{
+      "*": [
+        "./src/*.d.ts"
+      ]
+    }
   }
 }

--- a/packages/react/package.json
+++ b/packages/react/package.json
@@ -17,9 +17,9 @@
     "./jsx-runtime": {
       "import": "./src/jsx-runtime.mjs",
       "require": "./src/jsx-runtime.js",
-      "types":"./src/jsx-dev-runtime.d.ts"
+      "types":"./src/jsx-runtime.d.ts"
     },
-    "./jsx-dev-runtime.js": {
+    "./jsx-dev-runtime": {
       "import": "./src/jsx-dev-runtime.mjs",
       "require": "./src/jsx-dev-runtime.js",
       "types":"./src/jsx-dev-runtime.d.ts"

--- a/packages/react/src/jsx-namespace.d.ts
+++ b/packages/react/src/jsx-namespace.d.ts
@@ -4,6 +4,16 @@ import type { StylexProperty } from '@stylex-extend/shared'
 type ReactJSXLibraryManagedAttributes<C, P> = JSX.LibraryManagedAttributes<C, P>
 type ReactJSXIntrinsicElements = JSX.IntrinsicElements
 
+type ReactJSXElement = JSX.Element
+type ReactJSXElementClass = JSX.ElementClass
+type ReactJSXElementAttributesProperty = JSX.ElementAttributesProperty
+type ReactJSXElementChildrenAttribute = JSX.ElementChildrenAttribute
+type ReactJSXLibraryManagedAttributes<C, P> = JSX.LibraryManagedAttributes<C, P>
+type ReactJSXIntrinsicAttributes = JSX.IntrinsicAttributes
+type ReactJSXIntrinsicClassAttributes<T> = JSX.IntrinsicClassAttributes<T>
+type ReactJSXIntrinsicElements = JSX.IntrinsicElements
+type ReactJSXElementType = string | React.JSXElementConstructor<any>
+
 export namespace StyledJSX {
   export type LibraryManagedAttributes<C, P> = { stylex?: StylexProperty } & ReactJSXLibraryManagedAttributes<C, P>
 
@@ -12,4 +22,15 @@ export namespace StyledJSX {
       stylex?: StylexProperty
     }
   }
+
+  export type ElementType = ReactJSXElementType
+  export interface Element extends ReactJSXElement {}
+  export interface ElementClass extends ReactJSXElementClass {}
+  export interface ElementAttributesProperty
+    extends ReactJSXElementAttributesProperty {}
+  export interface ElementChildrenAttribute extends ReactJSXElementChildrenAttribute {}
+
+  export interface IntrinsicAttributes extends ReactJSXIntrinsicAttributes {}
+  export interface IntrinsicClassAttributes<T>
+    extends ReactJSXIntrinsicClassAttributes<T> {}
 }

